### ISSUE TITLE
python-3.6.0.tgz permissions borked

### ISSUE THREAD

**Issue by 12ed8076-2fdc-4989-adbb-4bf6d5dd549d** [python/cpython#73760]
BPO | [29574](https://bugs.python.org/issue29574)
--- | :---
Nosy | @malemburg, @ned-deily

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-03-23.00:35:14.111>
created_at = <Date 2017-02-15.22:50:47.223>
labels = ['build']
title = 'python-3.6.0.tgz permissions borked'
updated_at = <Date 2017-03-23.00:35:14.109>
user = 'https://bugs.python.org/DaveAnderson'
```

bugs.python.org fields:
```python
activity = <Date 2017-03-23.00:35:14.109>
actor = 'ned.deily'
assignee = 'none'
closed = True
closed_date = <Date 2017-03-23.00:35:14.111>
closer = 'ned.deily'
components = ['Build']
creation = <Date 2017-02-15.22:50:47.223>
creator = 'Dave_Anderson'
dependencies = []
files = []
hgrepos = []
issue_num = 29574
keywords = []
message_count = 4.0
messages = ['287891', '287892', '287918', '290019']
nosy_count = 3.0
nosy_names = ['lemburg', 'ned.deily', 'Dave_Anderson']
pr_nums = []
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = None
url = 'https://bugs.python.org/issue29574'
versions = ['Python 3.6']
```

</p></details>

**Comment by 12ed8076-2fdc-4989-adbb-4bf6d5dd549d** [comment]
Downloaded https://www.python.org/ftp/python/3.6.0/Python-3.6.0.tgz

Extracted on CentOS6 with sudo tar -xf Python-3.6.0.tgz

Result: 

[vagrant@developer tmp]$ ls -l Python-3.6.0
ls: cannot access Python-3.6.0/Tools: Permission denied
ls: cannot access Python-3.6.0/config.guess: Permission denied
ls: cannot access Python-3.6.0/pyconfig.h.in: Permission denied
ls: cannot access Python-3.6.0/config.sub: Permission denied
ls: cannot access Python-3.6.0/configure.ac: Permission denied
ls: cannot access Python-3.6.0/Python: Permission denied
ls: cannot access Python-3.6.0/Objects: Permission denied
ls: cannot access Python-3.6.0/PCbuild: Permission denied
ls: cannot access Python-3.6.0/Include: Permission denied
ls: cannot access Python-3.6.0/aclocal.m4: Permission denied
ls: cannot access Python-3.6.0/configure: Permission denied
ls: cannot access Python-3.6.0/Misc: Permission denied
ls: cannot access Python-3.6.0/Makefile.pre.in: Permission denied
ls: cannot access Python-3.6.0/setup.py: Permission denied
ls: cannot access Python-3.6.0/Lib: Permission denied
ls: cannot access Python-3.6.0/PC: Permission denied
ls: cannot access Python-3.6.0/Doc: Permission denied
ls: cannot access Python-3.6.0/README: Permission denied
ls: cannot access Python-3.6.0/Programs: Permission denied
ls: cannot access Python-3.6.0/install-sh: Permission denied
ls: cannot access Python-3.6.0/LICENSE: Permission denied
ls: cannot access Python-3.6.0/Modules: Permission denied
ls: cannot access Python-3.6.0/Grammar: Permission denied
ls: cannot access Python-3.6.0/Parser: Permission denied
ls: cannot access Python-3.6.0/Mac: Permission denied
total 0
-????????? ? ? ? ?            ? aclocal.m4
-????????? ? ? ? ?            ? config.guess
-????????? ? ? ? ?            ? config.sub
-????????? ? ? ? ?            ? configure
-????????? ? ? ? ?            ? configure.ac
d????????? ? ? ? ?            ? Doc
d????????? ? ? ? ?            ? Grammar
d????????? ? ? ? ?            ? Include
-????????? ? ? ? ?            ? install-sh
d????????? ? ? ? ?            ? Lib
-????????? ? ? ? ?            ? LICENSE
d????????? ? ? ? ?            ? Mac
-????????? ? ? ? ?            ? Makefile.pre.in
d????????? ? ? ? ?            ? Misc
d????????? ? ? ? ?            ? Modules
d????????? ? ? ? ?            ? Objects
d????????? ? ? ? ?            ? Parser
d????????? ? ? ? ?            ? PC
d????????? ? ? ? ?            ? PCbuild
d????????? ? ? ? ?            ? Programs
-????????? ? ? ? ?            ? pyconfig.h.in
d????????? ? ? ? ?            ? Python
-????????? ? ? ? ?            ? README
-????????? ? ? ? ?            ? setup.py
d????????? ? ? ? ?            ? Tools
[vagrant@developer tmp]$

Same operation with Python 3.5.2 tgz file downloaded from same location:

[vagrant@developer tmp]$ ls -l Python-3.5.2
total 1008
-rw-r--r--  1 1000 1000   8464 Jun 25  2016 aclocal.m4
-rwxr-xr-x  1 1000 1000  42856 Jun 25  2016 config.guess
-rwxr-xr-x  1 1000 1000  35740 Jun 25  2016 config.sub
-rwxr-xr-x  1 1000 1000 474932 Jun 25  2016 configure
-rw-r--r--  1 1000 1000 155069 Jun 25  2016 configure.ac
drwxrwxr-x 18 1000 1000   4096 Jun 25  2016 Doc
drwxrwxr-x  2 1000 1000   4096 Jun 25  2016 Grammar
drwxrwxr-x  2 1000 1000   4096 Jun 25  2016 Include
-rwxr-xr-x  1 1000 1000   7122 Jun 25  2016 install-sh
drwxrwxr-x 46 1000 1000  12288 Jun 25  2016 Lib
-rw-r--r--  1 1000 1000  12767 Jun 25  2016 LICENSE
drwxrwxr-x  8 1000 1000   4096 Jun 25  2016 Mac
-rw-r--r--  1 1000 1000  58449 Jun 25  2016 Makefile.pre.in
drwxrwxr-x  2 1000 1000   4096 Jun 25  2016 Misc
drwxrwxr-x 11 1000 1000   4096 Jun 25  2016 Modules
drwxrwxr-x  4 1000 1000   4096 Jun 25  2016 Objects
drwxrwxr-x  2 1000 1000   4096 Jun 25  2016 Parser
drwxrwxr-x  4 1000 1000   4096 Jun 25  2016 PC
drwxrwxr-x  2 1000 1000   4096 Jun 25  2016 PCbuild
drwxrwxr-x  2 1000 1000   4096 Jun 25  2016 Programs
-rw-r--r--  1 1000 1000  41897 Jun 25  2016 pyconfig.h.in
drwxrwxr-x  3 1000 1000   4096 Jun 25  2016 Python
-rw-r--r--  1 1000 1000   8060 Jun 25  2016 README
-rw-r--r--  1 1000 1000  99778 Jun 25  2016 setup.py
drwxrwxr-x 22 1000 1000   4096 Jun 25  2016 Tools
[vagrant@developer tmp]$

**Comment by 12ed8076-2fdc-4989-adbb-4bf6d5dd549d** [comment]
Sorry, should have shown sudo ls -l output for 3.6:

[vagrant@developer tmp]$ sudo ls -l Python-3.6.0
total 1016
-rw-r--r--  1 caturra games  10910 Dec 22 18:21 aclocal.m4
-rwxr-xr-x  1 caturra games  42856 Dec 22 18:21 config.guess
-rwxr-xr-x  1 caturra games  35740 Dec 22 18:21 config.sub
-rwxr-xr-x  1 caturra games 481627 Dec 22 18:21 configure
-rw-r--r--  1 caturra games 158661 Dec 22 18:21 configure.ac
drwxr--r-- 18 caturra games   4096 Dec 22 18:23 Doc
drwxr--r--  2 caturra games   4096 Dec 22 18:21 Grammar
drwxr--r--  2 caturra games   4096 Dec 22 18:21 Include
-rwxr-xr-x  1 caturra games   7122 Dec 22 18:21 install-sh
drwxr--r-- 33 caturra games   4096 Dec 22 18:21 Lib
-rw-r--r--  1 caturra games  12767 Dec 22 18:21 LICENSE
drwxr--r--  8 caturra games   4096 Dec 22 18:21 Mac
-rw-r--r--  1 caturra games  58829 Dec 22 18:21 Makefile.pre.in
drwxr--r--  2 caturra games   4096 Dec 22 18:21 Misc
drwxr--r-- 13 caturra games   4096 Dec 22 18:21 Modules
drwxr--r--  4 caturra games   4096 Dec 22 18:21 Objects
drwxr--r--  2 caturra games   4096 Dec 22 18:21 Parser
drwxr--r--  5 caturra games   4096 Dec 22 18:21 PC
drwxr--r--  2 caturra games   4096 Dec 22 18:21 PCbuild
drwxr--r--  2 caturra games   4096 Dec 22 18:21 Programs
-rw-r--r--  1 caturra games  41250 Dec 22 18:21 pyconfig.h.in
drwxr--r--  3 caturra games   4096 Dec 22 18:21 Python
-rw-r--r--  1 caturra games   8434 Dec 22 18:21 README
-rw-r--r--  1 caturra games 101041 Dec 22 18:21 setup.py
drwxr--r-- 24 caturra games   4096 Dec 22 18:21 Tools
[vagrant@developer tmp]$

**Comment by malemburg** [comment]
Indeed, there are two issue with the .tgz file:

 * it uses "staff" as group for all files (which will likely exist on some systems), but this appears unrelated in you case
 * all subdirs have go-x set, which prevents changing into the dir if you're not in the staff group, which is more of an issue

In your case, the system does not seem to have a staff group, but the numeric IDs stored in the .tgz file map to another user/group. As a result, you don't get access.

When creating .tgz files for redistribution, it's usually better to explicitly set the owner and group to either something that's not likely to exist on target machines or to root.root via --owner=root --group=root.

As user, you can work around this by using the options --no-same-owner --no-same-permissions when extracting the archive.

**Comment by ned-deily** [comment]
My apologies for the bad permissions in the tar file.  The newly-released 3.6.1 tar files should not have this anomaly.
